### PR TITLE
Refactor the code to make it easy for testing

### PR DESCRIPTION
**Pull request by Habush**
This PR moves out the fiber channels argument from the pattern matcher search functions to the outer annotation functions to make easy to test those functions. Also makes the parser testable and fixes the parser tests

### CODE REVIEW

**Comment by tanksha** [comment]
Where could this module be located?

https://github.com/Habush/annotation-scheme/blob/master/annotation/util.scm#L28

File: annotation/util.scm [hunk 1]
```diff
@@ -107,7 +108,7 @@
   Call (cog-execute! QUERY), return results, delete the SetLink.
   This avoids a memory leak of SetLinks
 "
-	; Run the query
+	Run the query
```
Comment (tanksha) [inline annotation/util.scm:111]: comment this line

File: annotation/util.scm [hunk 2]
```diff
@@ -25,6 +25,7 @@
 	#:use-module (opencog exec)
 	#:use-module (opencog bioscience)
 	#:use-module (annotation graph)
+  #:use-module (opencog grpc)
```
Comment (tanksha) [inline annotation/util.scm:28]: No code for module (opencog grpc)

Comment (tanksha) [reply annotation/util.scm:28]: Fixed the above issues and run make check 
PASS: tests/parser-tests.scm
PASS: tests/util-tests.scm
PASS: tests/pathway-tests.scm

Prints the above and takes forever, it didnt finish

I run each tests on its own and 
go-tests failed 
- (find-go-plus (ConceptNode "GO:1903896")) returns empty
main test failed at (length (parse-request (list "IGF1") "Dfaer" req))
- Wrong number of arguments to #<procedure parse-request (req)
...

Comment (Habush) [reply annotation/util.scm:28]: Removed the grpc module

**Comment by tanksha** [comment]
@Habush please fix the following
- There is an error which hangs the execution of main-tests, it is related to the parse-request function. The function accepts only one argument, two were given here https://github.com/Habush/annotation-scheme/blob/master/tests/main-tests.scm

I fixed that, but got another error
```
ERROR: In procedure string->symbol:
In procedure string->symbol: Wrong type argument in position 1 (expecting string): #f
```
Please try to run it first

- One test failed from go-tests, here is the log
```
test-name: go-regulates
location: tests/go-tests.scm:16
source:
+ (test-equal
+   "go-regulates"
+   3
+   (length
+     (find-go-plus (ConceptNode "GO:1903896"))))
expected-value: 3
actual-value: 0
result: FAIL
```
- The whole tests from string-test.scm failed with `unbound-variable` error please check them

**Comment by Habush** [comment]
@tanksha Fixed the test issues. Please check

**Pull request merged**